src/repr/src/row.rs

```diff
@@ -553,7 +553,7 @@ unsafe fn read_datum<'a>(data: &'a [u8], offset: &mut usize) -> Datum<'a> {
         Tag::Dummy => Datum::Dummy,
         Tag::Numeric => {
             let digits = read_byte(data, offset).into();
-            let exponent = read_byte(data, offset) as i8;
+            let exponent = i8::from_ne_bytes(read_byte(data, offset).to_ne_bytes());
```

Comment:
Yeah, so that it compiles into nothing no matter what the endianness of the target architecture is. Not that it really matters, since we only support x86. But I thought it was clearer this way. It's a C++ style `reinterpret_cast`.